File: algorithms_chapter_3/push_front.js
//create a function that inserts at the beginning of an arr, do this without
// built in functions

function push_front(arr, x){
  arr2 = []
  arr2[0] = x
  for (i = 0; i<arr.length; i++){
    arr2[i+1] = arr[i]
  }
  console.log(arr2)
}
array = [1,2,3,4,5,6,7]
push_front(array, 2)

File: algorithms_python/push_front.py
# create a function that pushes a value to the front of an array
# without creating a new array

array = [1,2,3,4,5]
def pushpop(arr, val):
    arr.append(val)
    for x in range(0,len(arr)-1):
        arr[x],arr[len(arr)-1] = arr[len(arr)-1],arr[x]
    print arr
    return arr

pushpop(array, 0)
